modules/wordpress-com-compose.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Module Name: WordPress.com Block Editor
```

Comment:
Is it necessary to create a separate module for this? (and one which is not automatically activated)
I think it would be simpler if this functionality was always included. That way we won't have to check the module activation status on WP.com side. What we can do instead is check for Jetpack version (which we'll probably have to do with module anyway). This also allows for easier handling of Atomic sites - we won't have to wait for Jetpack update to deploy our Calypso functionality.

We'll probably have some interference with Classic Block plugin that we'll need to handle sooner or later, and having one less toggle to worry about seems like a good thing. :)

Reply:
That would make sense. This is how Calypsoify works for example.

If you decide to go in that direction, I would recommend relying on the list we have here:
https://github.com/Automattic/jetpack/blob/593ee4cccd6aead047c72aa78378f368f084035e/modules/module-extras.php#L43

Once you add the module's entry point here, it will be loaded as soon as Jetpack is active and connected to WordPress.com.

You can then remove the different options from the Jetpack dashboard interface (in `_inc/client`) and `modules/module-info.php` since you won't need an interface anymore.

Reply:
> I think it would be simpler if this functionality was always included.

It _would_ be easier for all the reasons mentioned, but the big difference between this and Calypsoify is that we're making it possible by disabling a security-ish measure (silently removing the core header that intentionally blocks iframes); for this reason I thought an explicit activation by the user felt warranted. Although, since we're not explaining the headers specifically, maybe I'm misguided there – @xknown any arguments against having it silently enabled by default, with no action or knowledge by the user?

Reply:
Latest updates have this is an atuo-activated module.

Reply:
> It would be easier for all the reasons mentioned, but the big difference between this and Calypsoify is that we're making it possible by disabling a security-ish measure.

I thought about this - we are only removing the header in very (dotcom) specific circumstances in a (hopefully) secure way. To me it's seems like an implementation detail that users shouldn't have to worry about, since we are not compromising their security.

Reply:
Was this ever resolved? I also find it odd that it is a module; it seems like a pretty fundamental feature.

Can it just be part of Calypsoify?

Reply:
I've removed the module-ness from it; it should now auto-load for WordPress.com-connected sites, as suggested by @jeherve above 👍 

> Can it just be part of Calypsoify?

I didn't want to make it part of Calypsoify, no. Calypsoify's purpose was (is?) for local wp-admin pages to appear the same as Calypso pages on redirect, but it got all conflated in the rush for Gutenberg integration, and now is a mish-mash of various things. We plan to re-evaluate Calypsoify soon, so I'd rather keep this clear of it all.